xrtpy/response/xrt_teem.py

```diff
@@ -0,0 +1,735 @@
+        spatial binning factor
+
+    Trange : 2 element sequence containing floats [Optional]
+        Range of log10(temperature) values to examine. Must be in order from
+        lower to higher.
+
+    Te_err_threshold : float [Optional]
+        Threshold value for the temperature error. (default = 0.5 [50%])
+        Values of temperature with error exceeding the threshold are set to 0.
+
+    photon_noise_threshold : float [Optional]
+        Threshold value for ratio of photon noise to signal. (default = 0.2
+        [20%]). If photon noise/signal exceeds this value, the corresponding
+        temperatures are set to 0.
+
+    mask : Boolean array of the same shape as the images [Optional]
+        If provided, masks out parts of the images from the analysis. Note:
+        pixels to be masked out should be True, unmasked should be False
+
+    no_threshold : Boolean [Optional]
+        If True, no thresholds are set. (default = False)
+
+    verbose : Boolean [Optional]
+        If True, information is printed
+
+
+    Returns:
+    --------
+    T_e : ~sunpy.map.sources.hinode.XRTMap
+        image and metadata for log10 of the derived electron temperature [K].
+
+    EM : ~sunpy.map.sources.hinode.XRTMap
+        image and metadata for log10 of the derived volume emission measure [cm^-3].
+
+    T_error : ~sunpy.map.sources.hinode.XRTMap
+        image and metadata for uncertainty in log10 temperature [K].
+
+    EM_error : ~sunpy.map.sources.hinode.XRTMap
+        image and metadata for uncertainty in log10 volume emission measure [cm^-3].
+
+    Examples:
+    ---------
+    Using this function, you can derive the coronal temperature using
+    filter ratio method.
+
+    >>> T_e, EM, T_error, EMerror = xrt_teem(map1, map2) # doctest: +SKIP
+
+    If you want to bin the image data in space to reduce photon noise, set
+    binfac to the factor by which you want to bin.  For example to bin the
+    data by a factor of 3 do:
+
+    >>> T_e, EM, T_error, EMerror = xrt_teem(map1, map2, binfac=3) # doctest: +SKIP
+
+    The data is binned first and then the temperature is derived. Note that
+    the image size is reduced by the factor binfac in each dimension, which
+    contrasts with the behavior of the IDL routine xrt_teem.pro.
+
+    Notes
+    -----
+    The returned values of pixels where the temperature cannot be derived
+    or the error is greater than the threshold or the photon noise is
+    greater than the threshold are set to 0. The EM for those pixels is
+    also set to 0. The masks included in the output maps mask out pixels for
+    which the data for either image is 0 or for which temperature cannot be
+    determined, as well as for those masked out by an input mask (if
+    provided).
+
+    The details of the coronal-temperature-diagnostic capability of
+    Hinode/XRT is described in
+    Narukage et al. 2011, Solar Phys., 269, 169.
+    http://adsabs.harvard.edu/doi/10.1007/s11207-010-9685-2
```

Comment:
```suggestion
    :cite:t:`narukage:2011`
```
We could also do this for the article below if we added it to `docs/bibliography.bib`.